src/staticanalysis/bot/static_analysis_bot/workflows/remote.py

```diff
@@ -18,34 +18,37 @@ class AnalysisTask(object):
     '''
     An analysis CI task running on Taskcluster
     '''
-    def __init__(self, task_id):
+    def __init__(self, task_id, task_status):
+        assert 'task' in task_status
+        assert 'status' in task_status
         self.task_id = task_id
-        self.run_id = None
-        self.task = None
+        self.task = task_status['task']
+        self.status = task_status['status']
 
-    def load_logs(self, queue_service):
+    @property
+    def run_id(self):
+        return self.status['runs'][-1]['runId']
+
+    @property
+    def name(self):
+        return self.task['metadata'].get('name', 'unknown')
 
-        # Load base task
-        self.task = queue_service.task(self.task_id)
-        self.name = self.task['metadata'].get('name', 'unknown')
-        logger.info('Lookup task dependency', id=self.task_id, name=self.name)
+    @property
+    def state(self):
+        return self.status['state']
 
-        # Load task status
-        status = queue_service.status(self.task_id)
-        assert 'status' in status, 'No status data for {}'.format(self.task_id)
```

Comment:
Nit: Do you want to move this assertion to the constructor?

Reply:
:+1: